.github/workflows/python-build-and-lint.yml

```diff
@@ -0,0 +1,52 @@
+on:
+  workflow_call:
+
+jobs:
+
+  build_python_artifacts:
+    name: Build Artifacts
+    runs-on: ubuntu-22.04
+    defaults:
+      run:
+        working-directory: ./python
+    steps:
+      - name: Checkout Repository
+        uses: actions/checkout@main
+
+      - name: Set up Python 3.9.13
+        uses: actions/setup-python@main
+        with:
+          python-version: 3.9.13
+
+      - name: Install requirements
+        run: pip install -r requirements.txt
+
+      - name: Install the aac-puml Package
+        run: pip install -e .
```

Comment:
changed as requested

Reply:
The pip install performs the hash check of the modules in requirements.txt for additional security. I believe the dependencies section of the toml files has support for this, but it is in a different format that is output by the pip-compile tool that we are using to gather the hashes. If you want to consolidate these two steps into one, I recommend we schedule that work in a follow-on task for a future sprint.

Reply:
Also, doesn't there have to be at least one 'pip install' before the 'python build' command, in order to install the build module itself?

Reply:
The last comment is mixing manual implementation with what happens in the pipelines. While for the most part it is mirrored, it is not always because of how the actions are included and cover some steps we have to take manually. 

It was not necessary originally to have two pip install steps, but that was using a different build method with the `setup.py`, I am not entirely certain what all is impacted by changing to using the `toml` file still. Was part of why altering and improving the buildtools had been extracted as a separate and future feature for investigating and incorporating, but you jumped on it with this effort, so we'll find out some preliminary information. 

We can make additional adjustments as we learn more impacts.